### ISSUE TITLE
Training automatically restarts.How to fix it ?

### ISSUE THREAD

**Issue by sercan6843** [allenai/allennlp#2820]
**System:**
 - OS: Ubuntu 16.04
 - Python version: 3.6.1
 - AllenNLP version: v0.8.3

**Question**
Hello, i am working on a modified version of bidaf model.When a training ends allennlp starts a new one and removes everything from the first training.I am using default allennlp trainer.What might be the cause of that?Thanks in advance.

**Comment by brendan-ai2** [comment]
Hi @sercan6843. I'm not entirely sure I understand. It sounds like you're referring to how the trainer operates for multiple epochs. This is just another iteration over your dataset and is quite standard. Most models won't converge after only a single pass. You can control the maximum number of epochs here: https://github.com/allenai/allennlp/blob/master/training_config/bidaf.jsonnet#L89. But usually you'll be more interested in controlling the number of epochs via `patience` and `validation_metric`. See https://github.com/allenai/allennlp/blob/master/allennlp/training/trainer.py#L91 for details.

**Comment by sercan6843** [comment]
Hi @brendan-ai2 thanks for your answer ,but i wasnt referring to epochs.What i mean is after the patience ends for a training (i use -loss as metric), as soon as the program outputs final training accuracy,best validation acc. etc. it starts another training instead of just stopping.

**Comment by brendan-ai2** [comment]
Ah, sorry! That's quite strange! Could you please provide exactly what configuration you're using along with some logs around the time that the training loop repeats?

**Comment by sercan6843** [comment]
This is my trainer config.I tried a small training but it didnt happen this time.I have no idea why this happens but sometimes the training restarts from all over again.Thanks anyways @brendan-ai2 .If i face with this issue again i will try to provide some screenshots and logs.By the way i have 1 more question , for allennlp find-lr it should work just 100 iterations right?It restarts again too.I tried it and as soon as 1 find-lr ends it starts another instead of just stopping and after the second one ends it stops.
`  "trainer": {
    "num_epochs": 200,
    "grad_norm": 5.0,
    "patience": 10,
    "validation_metric": "-loss",
    "cuda_device": [0,1,2,3],
    "learning_rate_scheduler": {
      "type": "reduce_on_plateau",
      "factor": 0.5,
      "mode": "max",
      "patience": 2
    },
    "optimizer": {
      "type": "adam",
      "betas": [
        0.9,
        0.9
      ],
      "lr": 1e-4
    }
  }
}
`

**Comment by brendan-ai2** [comment]
Yes, `find-lr` shouldn't have this problem either. We suspect there's something strange going on with your shell and fairly confident this isn't an AllenNLP issue, but if you have any more conclusive logs, please do share. Thanks!
